src/eureka/S3_data_reduction/nircam.py

```diff
@@ -209,3 +229,159 @@ def cut_aperture(data, meta, log):
+        The metadata object.
+    log : logedit.Logedit
+        The current log.
+
+    Returns
+    -------
+    data : Xarray Dataset
+        The updated Dataset object with outlier background pixels flagged.
+    '''
+    log.writelog('  Performing outlier rejection...',
+                 mute=(not meta.verbose))
+
+    flux = data.flux.values
+    mask = data.mask.values
+    # FINDME: KBS removed estsig from inputs to speed up outlier detection.
+    # Need to test performance with and without estsig on real data.
+    if hasattr(meta, 'use_estsig') and meta.use_estsig:
+        bgerr = np.median(data.err)
+        estsig = [bgerr for j in range(len(meta.bg_thresh))]
+    else:
+        estsig = None
+
+    nbadpix_total = 0
+    for i in tqdm(range(flux.shape[1]), desc='Looping over Rows for outlier removal'):
+        for j in range(flux.shape[2]):  # Loops over Columns
+            ngoodpix = np.sum(mask[:, i, j] == 1)
+            data['mask'][:, i, j] *= sigrej.sigrej(flux[:, i, j], meta.bg_thresh,
+                                                   mask[:, i, j], estsig)
+            if not all(data['mask'][:, i, j].values):
+                # counting the amount of flagged bad pixels
+                nbadpix = ngoodpix - np.sum(data['mask'][:, i, j].values)
+                nbadpix_total += nbadpix
+    flag_percent = nbadpix_total/np.product(flux.shape)*100
+    log.writelog("  {0:.5f}% of the pixels have been flagged as outliers\n".format(flag_percent),
+                 mute=(not meta.verbose))
+
+    return data
+
+
+def corr_oof(data, meta, i, star_pos_x):
+    """
+    Correcting for 1/f noise.
+    """
+    print('Correcting for 1/f noise...')
+
+    # Let's first determine which amplifier regions are left in the frame.
+    # For NIRCam: 4 amplifiers, 512 pixels in x dimension per amplifier
+    pxl_idxs = np.arange(2048)  # Every NIRCam subarray has 2048 pixels in the x dimension
+    pxl_in_window_bool = np.zeros(2048, dtype=bool)
+    # pxl_in_window_bool is True for pixels which weren't trimmed away by meta.xwindow
+    for j in range(len(pxl_idxs)):
+        if meta.xwindow[0] <= pxl_idxs[j] < meta.xwindow[1]:
+            pxl_in_window_bool[j] = True
+    ampl_used_bool = np.array(list(map(any, pxl_in_window_bool.reshape((4, 512)))))
+    # Example: if only the middle two amplifier are left after trimming:
+    # ampl_used = [False, True, True, False]
+
+    star_pos_x_untrim = int(star_pos_x) + meta.xwindow[0]  # position of star before trimming
+    star_exclusion_area_untrim = \
+        np.array([star_pos_x_untrim - meta.oof_bg_dist, star_pos_x_untrim + meta.oof_bg_dist])
+
+    use_cols = np.ones(2048, dtype=bool)
+    for k in range(2048):
+        if star_exclusion_area_untrim[0] <= k < star_exclusion_area_untrim[1]:
+            use_cols[k] = False
+    use_cols = use_cols[meta.xwindow[0]:meta.xwindow[1]]
+    # Array with bools checking if column should be used for background subtraction
```

Comment:
I'm not exactly sure if I understand your point, sorry.
The reason I'm not doing this step in 2D is because the wings of the target on the detector are actually pretty big. So I have found that you shouldn't include anything closer than like 300 pixels from the centroid when you do the 1/f correction. Otherwise, you'd subtract stellar light in this step.
And because ny is way smaller than this area you should avoid, I'm only doing this in 1D.